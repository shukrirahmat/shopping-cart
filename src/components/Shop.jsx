import PropTypes from "prop-types";
import styles from "./Shop.module.css";

function Shop(props) {
  if (props.loading) {
    return (
      <div className={styles.wrapper}>
        <p className={styles.loadingText}>LOADING...</p>
      </div>
    );
  }

  return (
    <div className={styles.wrapper}>
      <div className={styles.itemContainer}>
        {props.items.map((item) => {
            return (
                <div className={styles.itemCard} key={item.id}>
                    <p>{item.title}</p>
                    <img src={item.image}></img>
                </div>
            )
        })}
      </div>
    </div>
  );
}

Shop.propTypes = {
  items: PropTypes.array,
  loading: PropTypes.bool,
};

export default Shop;
