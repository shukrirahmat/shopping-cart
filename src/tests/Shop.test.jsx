import { describe, it, expect } from "vitest";
import { render, screen } from "@testing-library/react";
import Shop from "../components/Shop.jsx";

describe("Shop component", () => {

    it("renders Shop page", () => {
        const {container} = render(<Shop items={[]} loading={false} addToCart={() => {}} />);
        expect(container).toMatchSnapshot();
    })

    it("show loading text", () => {
        render(<Shop items={[]} loading={true} addToCart={() => {}} />);
        const loadingtext = screen.getByRole('paragraph').textContent;
        expect(loadingtext).toMatch(/LOADING.../);
    })

})