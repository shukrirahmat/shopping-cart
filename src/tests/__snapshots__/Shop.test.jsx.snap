// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Shop component > renders Shop page 1`] = `
<div>
  <div
    class="_wrapper_1f04d7"
  >
    <div
      class="_category_1f04d7"
    >
      <p
        class="_categoryHead_1f04d7"
      >
        Men's Clothing
      </p>
      <div
        class="_itemsContainer_1f04d7"
      />
    </div>
    <div
      class="_category_1f04d7"
    >
      <p
        class="_categoryHead_1f04d7"
      >
        Women's Clothing
      </p>
      <div
        class="_itemsContainer_1f04d7"
      />
    </div>
    <div
      class="_category_1f04d7"
    >
      <p
        class="_categoryHead_1f04d7"
      >
        Jewelery
      </p>
      <div
        class="_itemsContainer_1f04d7"
      />
    </div>
    <div
      class="_category_1f04d7"
    >
      <p
        class="_categoryHead_1f04d7"
      >
        Electronics
      </p>
      <div
        class="_itemsContainer_1f04d7"
      />
    </div>
  </div>
</div>
`;
